<!--
  This file will launch four instances of the mmWave sensor node and configure each TI mmWave 6843 sensor using a 3D config. This launch file is intended
  to be used to launch the mmWave sensor node on TI's TDA4VM Jacinto Processors, as such a separate command must be used to launch rViz 
  for visualization on a PC. This File also launches mmWaveSync, a node which implements a user selected HW or SW based sync.
 -->

<launch>

  <node pkg="ti_mmwave_rospkg" type="mmWaveSync" name="mmWaveSync" ns="ti_mmwave_sync" output="screen">
    <param name="sync_type" value="SW" />
    <param name="num_sensors" value="4" />
    <param name="trigger_delay" value="0" />
    <param name="serial_ports" value="/dev/ttyUSB0 /dev/ttyUSB2 /dev/ttyUSB4 /dev/ttyUSB6" />
    <param name="config_file" value="$(find ti_mmwave_rospkg)/cfg/6843_3d.cfg" />
  </node>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="0" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyUSB0"/>
    <arg name="com_data" value="/dev/ttyUSB1"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="1" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyUSB2"/>
    <arg name="com_data" value="/dev/ttyUSB3"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="2" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyUSB4"/>
    <arg name="com_data" value="/dev/ttyUSB5"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="3" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyUSB6"/>
    <arg name="com_data" value="/dev/ttyUSB7"/>
  </include>
  
</launch>
